File: yggdrasil/examples/timed_pipe/src/timed_pipe_dst.R
library(yggdrasil)


run <- function() {
  print('Hello from R pipe_dst')

  # Ins/outs matching with the the model yaml
  inq <- YggInterface('YggInput', 'input_pipe')
  outf <- YggInterface('YggOutput', 'output_file')
  print("pipe_dst(R): Created I/O channels")

  # Continue receiving input from the queue
  count <- 0
  while (TRUE) {
    c(ret, buf) %<-% inq$recv()
    if (!ret) {
      print("pipe_dst(R): Input channel closed")
      break
    }
    ret <- outf$send(buf)
    if (!ret) {
      stop(sprintf("pipe_dst(R): SEND ERROR ON MSG %d", count))
    }
    count <- count + 1
  }

  fprintf('Goodbye from R destination. Received %d messages.', count)
}


run()

File: yggdrasil/examples/timed_pipe/src/timed_pipe_dst.jl
using Yggdrasil
using Printf


function run(args)
  println("Hello from Julia pipe_dst")

  # Ins/outs matching with the the model yaml
  inq = Yggdrasil.YggInterface("YggInput", "input_pipe")
  outf = Yggdrasil.YggInterface("YggOutput", "output_file")
  println("pipe_dst(Julia): Created I/O channels")

  # Continue receiving input from the queue
  global count = 0
  while (true)
    ret, buf = inq.recv()
    if (!ret)
      println("pipe_dst(Julia): Input channel closed")
      break
    end
    ret = outf.send(buf)
    if (!ret)
      error(@sprintf("pipe_dst(Julia): SEND ERROR ON MSG %d\n", count))
    end
    global count = count + 1
  end

  @printf("Goodbye from Julia destination. Received %d messages.\n", count)
end
    

run(ARGS)
